src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -82,6 +84,85 @@ enum MiscSpells
     SPELL_PET_NETHERWINDS_FATIGUED              = 160455
 };
 
+// 108281 - Ancestral Guidance
+class spell_sha_ancestral_guidance : public SpellScriptLoader
+{
+    public:
+        spell_sha_ancestral_guidance() : SpellScriptLoader("spell_sha_ancestral_guidance") { }
+
+        class spell_sha_ancestral_guidance_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_sha_ancestral_guidance_AuraScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_ANCESTRAL_GUIDANCE))
+                    return false;
+                if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_ANCESTRAL_GUIDANCE_HEAL))
+                    return false;
+                return true;
+            }
+
+            void HandleEffectProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+            {
+                PreventDefaultAction();
+                // No heal from this talent
+                if (eventInfo.GetHealInfo()->GetSpellInfo()->Id == SPELL_SHAMAN_ANCESTRAL_GUIDANCE_HEAL)
+                    return;
+                int32 bp0 = CalculatePct(int32(eventInfo.GetDamageInfo()->GetDamage() + eventInfo.GetHealInfo()->GetHeal()), aurEff->GetAmount());
+                if (bp0)
+                    eventInfo.GetActor()->CastCustomSpell(SPELL_SHAMAN_ANCESTRAL_GUIDANCE_HEAL, SPELLVALUE_BASE_POINT0, bp0, eventInfo.GetActor(), true, NULL, aurEff);
+            }
+
+            void Register() override
+            {
+                OnEffectProc += AuraEffectProcFn(spell_sha_ancestral_guidance_AuraScript::HandleEffectProc, EFFECT_0, SPELL_AURA_DUMMY);
+            }
+        };
+
+        AuraScript* GetAuraScript() const override
+        {
+            return new spell_sha_ancestral_guidance_AuraScript();
+        }
+};
+
+// 114911 - Ancestral Guidance Heal
+class spell_sha_ancestral_guidance_heal : public SpellScriptLoader
+{
+    public:
+        spell_sha_ancestral_guidance_heal() : SpellScriptLoader("spell_sha_ancestral_guidance_heal") { }
+
+        class spell_sha_ancestral_guidance_heal_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_sha_ancestral_guidance_heal_SpellScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_ANCESTRAL_GUIDANCE))
+                    return false;
+                if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_ANCESTRAL_GUIDANCE_HEAL))
+                    return false;
+                return true;
+            }
+
+            void ResizeTargets(std::list<WorldObject*>& targets)
+            {
+                if (targets.size() > 3)
```

Comment:
use Trinity::Containers::RandomResizeList (size check before calling it not needed)